native/src/main/java/io/ballerina/lib/data/yaml/parser/YamlParser.java

```diff
@@ -0,0 +1,2371 @@
+            }
+        }
+
+        if (tag.equals(DEFAULT_GLOBAL_TAG_HANDLE + "float")) {
+            if (state.schema == Types.YAMLSchema.JSON_SCHEMA) {
+                return TagResolutionUtils.constructSimpleFloat(data.toString(), state);
+            } else if (state.schema == Types.YAMLSchema.CORE_SCHEMA) {
+                return TagResolutionUtils.constructFloat(data.toString(), state);
+            }
+        }
+
+        if (tag.equals(DEFAULT_GLOBAL_TAG_HANDLE + "bool")) {
+            if (state.schema == Types.YAMLSchema.JSON_SCHEMA) {
+                return TagResolutionUtils.constructSimpleBool(data.toString(), state);
+            } else if (state.schema == Types.YAMLSchema.CORE_SCHEMA) {
+                return TagResolutionUtils.constructBool(data.toString(), state);
+            }
+        }
+
+        if (tag.equals(DEFAULT_GLOBAL_TAG_HANDLE + "null")) {
+            if (state.schema == Types.YAMLSchema.JSON_SCHEMA) {
+                return TagResolutionUtils.constructSimpleNull(data.toString(), state);
+            } else if (state.schema == Types.YAMLSchema.CORE_SCHEMA) {
+                return TagResolutionUtils.constructNull(data.toString(), state);
+            }
+        }
+
+        throw new Error.YamlParserException("tag schema not supported", state.getLine(), state.getColumn());
+    }
+
+    /**
+     * Obtain an event for a set of tokens.
+     *
+     * @param state - Current parser state
+     * @param docType - Document type to be parsed
+     * @return - Parsed event
+     */
+    private static YamlEvent handleEvent(ComposerState state, Types.DocumentType docType)
+            throws Error.YamlParserException {
+        if (state.terminatedDocEvent != null &&
+                state.terminatedDocEvent.getKind() == YamlEvent.EventKind.DOCUMENT_MARKER_EVENT) {
+            return state.terminatedDocEvent;
+        }
+        return parse(state.parserState, ParserUtils.ParserOption.DEFAULT, docType);
+    }
+
+    /**
+     * Obtain an event for a set of tokens.
+     *
+     * @param state - Current parser state
+     * @param option - Expected values inside a mapping collection
+     * @return - Parsed event
+     */
+    private static YamlEvent handleEvent(ComposerState state, ParserUtils.ParserOption option)
+            throws Error.YamlParserException {
+        if (state.terminatedDocEvent != null &&
+                state.terminatedDocEvent.getKind() == YamlEvent.EventKind.DOCUMENT_MARKER_EVENT) {
+            return state.terminatedDocEvent;
+        }
+        return parse(state.parserState, option, BARE_DOCUMENT);
+    }
+
+    /**
+     * Obtain an event for a set of tokens.
+     *
+     * @param state - Current parser state
+     * @param option - Expected values inside a mapping collection
+     * @param docType - Document type to be parsed
+     * @return - Parsed event
+     */
+    private static YamlEvent parse(ParserState state, ParserUtils.ParserOption option,
```

Comment:
This method seems quit large. Shall we move part of them into separate methods? WDYT?

Reply:
Will refactor in a separate PR.